modules/visual_script/visual_script_property_selector.cpp

```diff
@@ -212,8 +212,8 @@ void VisualScriptPropertySelector::_update_search() {
 			}
 
 			if (input == String() ||
-					set_text_raw.findn(input) != -1 &&
-							set_text.findn(input) != -1) {
+					(set_text_raw.findn(input) != -1 &&
+							set_text.findn(input) != -1)) {
```

Comment:
@fire Please review the above two changes. I assume this clarifies the *intent*, yet I'm not sure if the code is correct in the first place, both `if` conditions seem pretty similar (and hard to understand) and the second one seems to override the item 0 set by the first one.

Reply:
The first part. The conditions trigger if A) the string is empty B) input is in the text C) input is in the capitalized() text, when true, so I'm not sure what grouping means here.

The second part. This code generates the list of search items. So the second time doesn't overwrite the first because  `TreeItem *item = search_options->create_item(category ? category : root);` creates a new item.

Reply:
> The conditions trigger if A) the string is empty B) input is in the text C) input is in the capitalized() text, when true, so I'm not sure what grouping means here.

Well grouping is necessary since the condition is written as `A || B && C`, and `(A || B) && CC` is different from `A || (B && C)`.

Reply:
Also note that the first `if` uses `A || B || C`.

Reply:
Logically A || B || C is correct. Add the item to the tree if the input is empty or if the input is found in either raw string or input is found in the capitalized() string. I got confused a bit.

note: not (not found) is the same as found

Reply:
I'll amend the commit to use `||` both times then.

BTW, `findn` is case insensitive so does this `.capitalize()` business and check of both raw and capitalized string make sense? It sounds like one single check should be enough.

Reply:
capitalize also converts "A Function" to "a_function"

Reply:
String capitalize ( )

> Changes the case of some letters. Replaces underscores with spaces, converts all letters to lowercase, then capitalizes first and every letter following the space character. For capitalize camelCase mixed_with_underscores it will return Capitalize Camelcase Mixed With Underscores.

Reply:
See https://github.com/godotengine/godot/pull/22446/commits/936972a770b29de78f611adf2bf354f2858c27af#diff-70a18288c4c984bbb1711423dfd08164R193

Reply:
https://github.com/godotengine/godot/pull/22446/commits/4c5917baca0d9f3c6be554d0c5077c578ee7b3de#diff-70a18288c4c984bbb1711423dfd08164R193 should be correct now.